src/Umbraco.Infrastructure/Media/ImageSharpImageService.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.IO;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Processing;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Infrastructure.Media
+{
+    public class ImageSharpImageService : IImageSharpImageService
+    {
+
+        public ImageFileModel GetImage(string imageOriginalPath, int? width, int? height)
+        {
+            byte[] imageData;
+            IImageFormat imageFormat;
+
+            if (string.IsNullOrEmpty(imageOriginalPath)) return null;
+
+            using (var image = Image.Load(imageOriginalPath, out imageFormat))
+            {
+                image.Mutate(x =>
+                {
+                    if (width != null)
+                        if (height != null)
+                            x.Resize((int) width, (int) height);
```

Comment:
You want to avoid null checks in your interface design. Each implementation now requires null checks.